pydomcfg/domzgr/zgr.py

```diff
@@ -0,0 +1,154 @@
+"""
+Base class to generate NEMO v4.0 vertical grids.
+"""
+
+from itertools import product
+
+import numpy as np
+from xarray import DataArray, Dataset
+
+
+class Zgr:
+    """
+    Base class to generate NEMO vertical grids.
+    """
+
+    def __init__(self, ds_bathy: Dataset, jpk: int):
+        """
+        Initialize class.
+
+        Parameters
+        ----------
+        ds_bathy: Dataset
+             xarray dataset including grid coordinates and bathymetry
+        jpk: int
+             number of computational levels
+        """
+
+        self._bathy = ds_bathy
+        self._jpk = jpk
+
+    # -------------------------------------------------------------------------
+    def init_ds(self) -> Dataset:
+        ds = self._bathy.copy()
+        jpi = ds.dims["x"]
+        jpj = ds.dims["y"]
+        jpk = self._jpk
+
+        var = ["z3", "e3"]
+        grd = ["T", "W"]
+        crd = [("z", range(jpk)), ("y", range(jpj)), ("x", range(jpi))]
+
+        # Initialise a dataset with z3 and e3 dataarray filled with nan
+        da = DataArray(np.zeros(shape=(jpk, jpj, jpi)) * np.nan, coords=crd)
```

Comment:
We need to avoid this kind of initialization. We don't know if the user is using pure `numpy` or `dask`, but this creates a 3D numpy array and loads it into memory.
I think you can do something like this:
```python
da = xr.full_like(ds["any2Dvariable"], None).expand_dims(z=range(jpk))
```

Reply:
Ok, I see your point! Thanks!